stretch_core/stretch_core/joint_trajectory_server.py

```diff
@@ -108,12 +109,14 @@ def execute_cb(self, goal_handle):
             pickle.dump(goal, s)
 
         # Register this goal's ID
-        goal_id = self.latest_goal_id + 1
-        self.latest_goal_id += 1
+        goal_id = self.latest_goal_id
         
         with self.node.robot_stop_lock:
             # Escape stopped mode to execute trajectory
             self.node.stop_the_robot = False
+        # TODO: To avoid bugs (e.g., where we return and forget to release the lock),
```

Comment:
Good idea. Please start a Github issue for this.

Reply:
Done. https://github.com/hello-robot/stretch_ros2/issues/152